### PR TITLE
fix precedence issue with facts

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

avoid making gathered facts high precedence, only set_fact is supposed to be.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

facts
##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

**Comment by alikins** [comment]
Haven't tested yet, but does this avoid the issue mentioned in https://github.com/ansible/ansible/pull/26153#issuecomment-311708915 ?  


> A fact set non caching, and then later set with caching, the non caching value wins because of the higher precedence of non_persistent facts. Adding to non persistent facts and host_facts seems to DWIM there, but unsure of consequences elsewhere.

**Comment by alikins** [comment]
some intg tests at https://github.com/ansible/ansible/compare/devel...alikins:set_fact_prec_intg_tests_32302 if you want to include them

**Comment by alikins** [comment]
LGTM if the test cases mentioned above cover this case (they pass with this branch)

File: docs/docsite/rst/playbooks_variables.rst [hunk 1]
```diff
@@ -884,6 +884,8 @@ Basically, anything that goes into "role defaults" (the defaults folder inside t
 
 .. [1] Tasks in each role will see their own role's defaults. Tasks defined outside of a role will see the last role's defaults.
 .. [2] Variables defined in inventory file or provided by dynamic inventory.
+.. [3] When created with set_facts's cacheable option, variables will have the high precedence in the play,
```
Comment (alikins) [inline docs/docsite/rst/playbooks_variables.rst:887]: Should this be multiple levels?

* host facts
* variables created with cacheable set_facts in a previous play when read from the cache
<..>
*  variables created with set_facts in current play

Is there anything else to distinguish the precendence of host_facts  and 'cached set_facts'?

Comment (bcoca) [reply docs/docsite/rst/playbooks_variables.rst:887]: no, they update the same dict, it is order in which they were set, but not by their nature

**Comment by abadger** [comment]
The tests are now passing for this change  (The failure is unstable due to a fedora 24 package repo) so unrelated to our code.

@dharmabumstead there's a bit of docs in here that are ready for your perusal.

**Comment by dharmabumstead** [comment]
Reminder: please don’t merge PRs with doc updates until the docs team as reviewed and approved. Thanks.

**Comment by abadger** [comment]
@dharmabumstead Did this PR need any additional changes?  I'm going to cherry-pick it for 2.4 and want to make sure I get any additional fixes you might have had.

**Comment by abadger** [comment]
dhamabumstead confirmed docs are okay to cherrypick on slack

Cherrypicked to stable-2.4 for the 2.4.2beta2 release.

**Pull request merged**